### PR TITLE
Use setuptools instead of distutils for Python > 3.8.

### LINKED ISSUE #2112: Build failure with python3.12
From Fedora 39 which has python3.12 by default
https://bugzilla.redhat.com/show_bug.cgi?id=2226397

```
Traceback (most recent call last):
  File "/builddir/build/BUILD/rnp-v0.17.0/src/tests/cli_tests.py", line 14, in <module>
    from cli_common import (file_text, find_utility, is_windows, list_upto,
  File "/builddir/build/BUILD/rnp-v0.17.0/src/tests/cli_common.py", line 2, in <module>
    import distutils.spawn
ModuleNotFoundError: No module named 'distutils'
CMake Error at src/tests/CMakeLists.txt:237 (message):
  Failed to retrieve test suite list for cli_tests

```

Switching to `setuptools._distutils.spawn` seems a working solution, but is only working with recent python versions
Tested on RHEL-9 with python 3.9 and setuptools 53.0

### PR DESCRIPTION
**Pull request by ni4**
Fixes #2112

### CODE REVIEW

**Comment by remicollet** [comment]
Is this enough (seems to work)

Shouldn't this be also changed ?
`path = distutils.spawn.find_executable(name)`

**Comment by remicollet** [comment]
I confirm this patch works as expected

**Review by ronaldtse — APPROVED** [review]

**Comment by ni4** [comment]
@remicollet Thanks for confirming. Wondering how that worked without `find_executable`. Please see the updated patch.

File: src/tests/cli_common.py [hunk 1]
```diff
@@ -78,7 +82,7 @@ def file_text(path, encoding = CONSOLE_ENCODING):
         return f.read().decode(encoding).replace('\r\r', '\r')
 
 def find_utility(name, exitifnone = True):
-    path = distutils.spawn.find_executable(name)
+    path = shutil.which(name)
```
Comment (remicollet) [inline src/tests/cli_common.py:85]: Not tested, but `setuptools._distutils.spawn` provides distutils.spawn and find_executable.
```
# head /usr/lib/python3.11/site-packages/setuptools/_distutils/spawn.py
"""distutils.spawn

Provides the 'spawn()' function, a front-end to various platform-
specific functions for launching another program in a sub-process.
Also provides the 'find_executable()' to search the path for a given
executable name.
"""

```

File: src/tests/cli_common.py [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-import distutils.spawn
 import random
 import string
 import logging
 import os
 import re
+import shutil
 from subprocess import Popen, PIPE
+if sys.version_info < (3, 12):
+    import distutils.spawn
```
[the remark this answers is not in the document]
Comment (remicollet) [reply src/tests/cli_common.py:10]: using shutil make  distutils unused

**Comment by remicollet** [comment]
Tested with various python versions (Fedora, RHEL-8 and 9) using shlib (and removing import of distutils)
Build OK.

**Comment by ronaldtse** [comment]
@ni4 other than the comments from @remicollet there is also a failure in Cirrus:

https://cirrus-ci.com/task/6093471186944000
```
The following tests FAILED:
	258 - cli_tests-Encryption (Failed)
Errors while running CTest
```

File: src/tests/cli_common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ni4) [reply src/tests/cli_common.py]: Agree, thanks.

File: src/tests/cli_common.py (hunk 1 above)
Comment (ni4) [reply src/tests/cli_common.py:81]: Thanks. Let's use `shutil.which` as it seems to be easier solution. Or do you know any disadvantages?

Comment (remicollet) [reply src/tests/cli_common.py:81]: Yes  shutil.which seems the simpler.

LGTM

**Comment by ronaldtse** [comment]
Ping @antonsviridenko for approval. Thanks!

**Comment by ni4** [comment]
@maxirmx Do you have an idea why Windows-2019 OpenSSL runners fail? I added some debug logging, but those doesn't given any idea to me.

**Comment by maxirmx** [comment]
@ni4 
It looks like there is some critical issue in between vcpkg and nasm: https://github.com/rnpgp/rnp/actions/runs/5700501434  

It does not look like the issue you mentioned but we have to wait until it is resolved

**Comment by antonsviridenko** [comment]
```
 -- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR) 
CMake Error at C:/Program Files/CMake/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR) (Required is at least version "1.1.1")
Call Stack (most recent call first):
-- Configuring incomplete, errors occurred!
  C:/Program Files/CMake/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.27/Modules/FindOpenSSL.cmake:668 (find_package_handle_standard_args)
  src/lib/CMakeLists.txt:44 (find_package)
```

**Review by antonsviridenko — APPROVED** [review]
@ronaldtse I'm not an expert at Python, but otherwise looks good.

**Comment by maxirmx** [comment]
https://github.com/microsoft/vcpkg/issues/32600
Our scripts were operating from hash but hash has been updated in line with vcpkg
So now Windows/openssl is totally broken

- I suggest to disable MSVC/openssl build for now
- If this issue persists we can consider other openssl Windows distribution. [Shining Light](https://slproweb.com/products/Win32OpenSSL.html) or something similar

**Comment by ni4** [comment]
@maxirmx Thanks for the investigation!
@ronaldtse Should we merge this as is, given that patch doesn't have nothing to deal with OpenSSL/Windows?

**Comment by ronaldtse** [comment]
Agree @ni4 , let’s merge first. Thanks!

**Pull request merged**